agent/consul/server.go

```diff
@@ -858,11 +861,14 @@ func NewServer(config *Config, flat Deps, externalGRPCServer *grpc.Server, incom
 	return s, nil
 }
 
-func (s *Server) registerResources() {
-	catalog.RegisterTypes(s.typeRegistry)
-	catalog.RegisterControllers(s.controllerManager, catalog.DefaultControllerDependencies())
+func (s *Server) registerResources(deps Deps) {
+	if stringslice.Contains(deps.Experiments, catalogResourceExperimentName) {
+		catalog.RegisterTypes(s.typeRegistry)
+		catalog.RegisterControllers(s.controllerManager, catalog.DefaultControllerDependencies())
+
+		mesh.RegisterTypes(s.typeRegistry)
```

Comment:
Because this also registers mesh types, should make the feature flag more generic, like `v2-apis` or `resource-apis`?

I'm thinking that we can re-use it for future v2 stuff without needed to provide multiple because they will all depend on each other. For example, if you're enabling mesh you'd also want to enable catalog.

Reply:
Any suggestions for the name?

Reply:
What do you think about `v2-apis` or `resource-apis`? Is it too broad?

Reply:
`resource-apis` it is.